src/plugins/vis_type_table/public/to_ast.ts

```diff
@@ -23,6 +23,7 @@ export const toExpressionAst = (vis: Vis, params: any) => {
   const schemas = getVisSchemas(vis, params);
 
   const tableData = {
+    title: vis.title,
```

Comment:
Is visualization title always present or do we need some default title set?

Reply:
good question. Here table vis is using Visualization plugin. title is initiated [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/src/plugins/visualizations/public/vis.ts#L100)

```
public title: string = '';
```

Reply:
The only place requires initialization is `buckets: schemas.bucket || []`. This is because for table vis, we defined  buckets as a required field in `TableVisConfig`:

```
export interface TableVisConfig extends TableVisParams {
  title: string;
  metrics: SchemaConfig[];
  buckets: SchemaConfig[];
  splitRow?: SchemaConfig[];
  splitColumn?: SchemaConfig[];
}
```

But in Visualization plugin, `bucket` is defined as `any[] || undefined` since it is optional, in Schema. Therefore, we need to initiated to `[]`. 

```
export interface Schemas {
  metric: SchemaConfig[];
  bucket?: any[];
  geo_centroid?: any[];
  group?: any[];
  params?: any[];
  radius?: any[];
  segment?: any[];
  split_column?: any[];
  split_row?: any[];
  width?: any[];
  // catch all for schema name
  [key: string]: any[] | undefined;
}
```

Reply:
> SchemaConfig

I see! Is there a specific reason as to why `buckets` is a required field in Table vis?